contracts/account/utils/ERC6900Utils.sol

```diff
@@ -0,0 +1,124 @@
+    error ERC6900ExecutionSelectorNotAllowedForGlobalValidation(
+        ModuleEntity validationModuleEntity,
+        bytes4 executionSelector
+    );
+    error ERC6900InvalidExecuteTarget();
+    error ERC6900ExecutionFailed(address target, uint256 value, bytes data);
+
+    // ModuleEntity
+
+    function unpack(ModuleEntity moduleEntity) internal pure returns (address, uint32) {
+        return (module(moduleEntity), entityId(moduleEntity));
+    }
+
+    function module(ModuleEntity moduleEntity) internal pure returns (address) {
+        return address(Packing.extract_32_20(ModuleEntity.unwrap(moduleEntity), 0));
+    }
+
+    function entityId(ModuleEntity moduleEntity) internal pure returns (uint32) {
+        return uint32(Packing.extract_32_4(ModuleEntity.unwrap(moduleEntity), 20));
+    }
+
+    // ValidationFlags
+
+    function isGlobalValidation(ValidationFlags validationFlags) internal pure returns (bool) {
+        return ValidationFlags.unwrap(validationFlags) & uint8(0x04) == 0x04; // 0b00000100
+    }
+
+    function isSignatureValidation(ValidationFlags validationFlags) internal pure returns (bool) {
+        return ValidationFlags.unwrap(validationFlags) & uint8(0x02) == 0x02; // 0b00000010
+    }
+
+    function isUserOpValidation(ValidationFlags validationFlags) internal pure returns (bool) {
+        return ValidationFlags.unwrap(validationFlags) & uint8(0x01) == 0x01; // 0b00000001
+    }
+
+    // ValidationConfig
+
+    // function module(ValidationConfig validationConfig) internal pure returns (address) {
+    //     return address(Packing.extract_32_20(ValidationConfig.unwrap(validationConfig), 0));
+    // }
+
+    // function module(bytes calldata hook) internal pure returns (address) {
+    //     return address(Packing.extract_32_20(bytes32(hook), 0));
+    // }
+
+    // function entityId(ValidationConfig validationConfig) internal pure returns (uint32) {
+    //     return uint32(Packing.extract_32_4(ValidationConfig.unwrap(validationConfig), 20));
+    // }
+
+    function unpack(ValidationConfig validationConfig) internal pure returns (ModuleEntity, ValidationFlags) {
+        return (
+            ModuleEntity.wrap(Packing.extract_32_24(ValidationConfig.unwrap(validationConfig), 0)),
+            ValidationFlags.wrap(uint8(Packing.extract_32_1(ValidationConfig.unwrap(validationConfig), 21)))
+        );
+    }
+
+    // function moduleEntity(ValidationConfig validationConfig) internal pure returns (ModuleEntity) {
+    //     return ModuleEntity.wrap(Packing.extract_32_24(ValidationConfig.unwrap(validationConfig), 0));
+    // }
+
+    // function getValidationFlags(ValidationConfig validationConfig) internal pure returns (ValidationFlags) {
+    //     return ValidationFlags.wrap(uint8(Packing.extract_32_1(ValidationConfig.unwrap(validationConfig), 21)));
+    // }
+
+    // HookConfig
+
+    function extractHookConfig(bytes calldata hook) internal pure returns (HookConfig) {
```

Comment:
It feels like `getHookData` and `extractHookConfig` are duplicated code. Can we get rid of one?

Reply:
Yes see [Remove duplicated HookConfig extractor](https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80/commits/078c232c8f5437fc0d864810ffad72a391de5444).